File: vol1/ch5/LeetCode41-1.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        hset=set()
        for x in nums: hset.add(x)
        ans=1
        while ans<=len(nums)+1:
            if ans not in hset:
                break;
            ans+=1
        return ans

File: vol1/ch5/LeetCode41-1.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        unordered_set<int> hset;
        for(int x:nums) hset.insert(x);
        int ans=1;
        while(ans<=nums.size()+1) {
            if(hset.find(ans)==hset.end())
                break;
            ans++;
        }
        return ans;
    }
};

